lib/spack/spack/util/file_cache.py

```diff
@@ -170,7 +170,8 @@ def remove(self, key):
             os.unlink(self.cache_path(key))
         finally:
             lock.release_write()
-        os.unlink(self._lock_path(key))
+        if lock._enable:
```

Comment:
Could you add a `spack.util.lock.Lock.remove` function, which does nothing when `_enable` is `False`? Whenever possible it is ideal to avoid referring to private members of classes.